File: src/main/java/com/hjj/apiserver/service/ScheduleService.java
package com.hjj.apiserver.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.time.LocalDateTime;

@Slf4j
@Component
@EnableAsync
@RequiredArgsConstructor
public class ScheduleService {
    private final WebScrappingService webScrappingService;

    @Scheduled(fixedRate = 100000, initialDelay = 5000)
    public void scheduleliivMateTodayQuizTask(){
        try{
            log.info("scheduleliivMateTodayQuizTask: {}", LocalDateTime.now());
            webScrappingService.liivMateTodayQuizAnswerFind();
        }catch (Exception e){
            log.error("scheduleliivMateTodayQuizTask: {}", e);
        }

    }
}

File: src/main/kotlin/com/hjj/apiserver/service/impl/ScheduleService.kt
package com.hjj.apiserver.service.impl

import com.hjj.apiserver.util.logger
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.time.LocalDateTime

@Component
class ScheduleService(
    private val webScrappingService: WebScrappingService,
) {
    private val log = logger()

    @Scheduled(cron = "0 00 14 * * *")
    fun scheduleliivMateTodayQuizTask() {
        try {
            log.info("scheduleliivMateTodayQuizTask: {}", LocalDateTime.now())
            webScrappingService.liivMateTodayQuizAnswerFind()
        } catch (e: Exception) {
            log.error("scheduleliivMateTodayQuizTask: {}", e)
        }
    }
}
